packages/reports/addon/components/navi-column-config.ts

```diff
@@ -0,0 +1,175 @@
+import { easeOut, easeIn } from 'ember-animated/easings/cosine';
+import { inject as service } from '@ember/service';
+import ReportModel from 'navi-core/models/report';
+import ColumnFragment from 'navi-core/models/bard-request-v2/fragments/column';
+import { Parameters, ColumnType } from 'navi-data/adapters/facts/interface';
+import { tracked } from '@glimmer/tracking';
+import ColumnMetadataModel from 'navi-data/models/metadata/column';
+
+interface NaviColumnConfigArgs {
+  isOpen: boolean;
+  report: ReportModel;
+  lastAddedColumn: ColumnFragment;
+  onAddColumn(metadata: ColumnMetadataModel, parameters: Parameters): void;
+  onRemoveColumn(metadata: ColumnMetadataModel, parameters: Parameters): void;
+  onToggleFilter(column: ColumnFragment): void;
+  onRenameColumn(column: ColumnFragment, alias: string): void;
+  onReorderColumn(column: ColumnFragment, index: number): void;
+  openFilters(): void;
+  drawerDidChange(): void;
+}
+
+type ConfigColumn = {
+  type: ColumnType;
+  name: string;
+  displayName: string;
+  isFiltered: boolean;
+  isRemovable: boolean;
+  fragment: ColumnFragment;
+};
+
+export default class NaviColumnConfig extends Component<NaviColumnConfigArgs> {
+  /**
+   * Dimension and metric columns from the request
+   */
+  @computed('args.report.request.{columns.@each.parameters,filters.[]}')
+  get columns(): ConfigColumn[] {
+    const { request } = this.args.report;
+    if (request.table !== undefined) {
+      const { metricColumns: metrics, dimensionColumns: dimensions, dimensionFilters, metricFilters } = request;
+
+      const filteredDimensions = dimensionFilters.map(({ canonicalName }) => canonicalName);
+      const filteredMetrics = metricFilters.map(({ canonicalName }) => canonicalName);
+      const dimensionColumns = dimensions.map(dimension => {
+        const { canonicalName: name, type } = dimension;
+        return {
+          type,
+          name,
+          displayName: this.getDisplayName(dimension),
+          isFiltered: filteredDimensions.includes(name),
+          isRemovable: true,
+          fragment: dimension
+        };
+      });
+
+      const metricColumns = metrics.map(metric => {
+        const { canonicalName: name, type } = metric;
+        return {
+          type,
+          name,
+          displayName: this.getDisplayName(metric),
+          isFiltered: filteredMetrics.includes(name),
+          isRemovable: true,
+          fragment: metric
+        };
+      });
+
+      return [...dimensionColumns, ...metricColumns];
+    }
+    return [];
+  }
+
+  @service metricName: TODO;
+
+  @tracked
+  currentlyOpenColumn?: ConfigColumn;
+
+  @tracked
+  componentElement!: Element;
+
+  getDisplayName(column: ColumnFragment) {
+    const {
+      alias,
+      columnMetadata: { name },
+      hasParametersSet,
+      parameters
+    } = column;
+
+    if (alias) {
+      return alias;
+    }
+    return hasParametersSet ? `${name} (${Object.values(parameters).join(' ')})` : name;
```

Comment:
Could replace this with `naviFormatter.formatMetric(columnMetadata, parameters, alias)` (needs a rename)